galley/pkg/config/resource/name.go

```diff
@@ -52,6 +57,18 @@ func NewFullName(name string) (Name, error) {
 	return Name{string: name}, nil
 }
 
+// NewFullOrLocalName returns a given name as a resource Name.  If name was a local name, the supplied namespace is used
+func NewFullOrLocalName(namespace, name string) Name {
+	if strings.ContainsRune(name, '/') {
+		if retval, err := NewFullName(name); err == nil {
+			return retval
+		}
+		return InvalidName
```

Comment:
I think I prefer the (result, err) return like NewFullName. It's self-consistent within the package and also more consistent with Go style.